### ISSUE TITLE
a11y: Address structural issues and labels with Key/Value control 

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12813]
a11y issue: 193, 188

**Issue 193 - Data table not programmatically defined**

The mentioned content is visually presented as a data table but not marked-up as table programatically resulting in screen reader users finding it difficult to understand the content structure.

Occurrences to be fixed:

Repository: Create
- Key 
- Value 
- e.g. foo and e.g. bar input fields (available on activating "Add Label" and "Add Annotation" buttons).
- Remove and so on

Home Links
- Display Text 
- URL
- e.g. foo and e.g. bar input fields (available on activating "Add Link" button).
- Remove
- Docs
- https://ranchermanager.docs.rancher.com/v2.9 and so on

Example:

![image](https://github.com/user-attachments/assets/434983e7-86fb-4e11-8363-9e13caa598ab)

Suggested remedition:

Utilize table mark-up, such as <table>, <tr>, <th>, <td>,  and so on to present the content as a data table.

Issue 188 - Missing form label or 'title' attribute

The form fields do not have the label associated with them and "title" attribute is not included in their source code resulting in screen reader users will find it difficult to understand the purpose of the form fields.

Occurrences:

Repository: Create
- Key and Value input fields (available on activating "Add Label" and "Add Annotation" buttons).

Git Repo - Add Repository
-  Path (e.g. /directory/in/your/repo) input fields (available on activating Add Path button)

Home Links
- Display Text  and URL e.g. foo input fields (available on activating "Add Link" buttons)

Example:

![image](https://github.com/user-attachments/assets/27da154f-c1f0-47c1-934c-d6a43e3bc61a)

Suggested remediation:

Provide unique and descriptive labels such as "Key 1", "Key 2", "Key 3", "Value 1", "Value 2", "Value 3" and so on for the mentioned input fields using "aria-label" attribute.

**Comment by yonasberhe23** [comment]
Requires manual validation.

**Comment by izaac** [comment]
the `area-label` property value is gettng created like this: `%generic.ariaLabel.value(index: 0)%`
Reopened.

```html
<textarea data-testid="value-multiline" placeholder="e.g. bar" class="no-resize no-ease" aria-label="%generic.ariaLabel.value(index: 0)%" spellcheck="false" value="" style="height: 40px; overflow: hidden;"></textarea>
```

**Comment by izaac** [comment]
Rancher version: `v2.11-a2ff78325d1768b0a52fa65a88c00a1f3cddace5-head`
UI Version: `master 8e2d825`
Firefox version: `135.0.1`

Key Value areas validated:

- Continuous Delivery -> Add repository and Create.
- Global Settings > Home Links

Labels and values can be set and CRUD operations work as expected.
`aria-label` properties are now fixed too. 
✅
